device/cuda/src/clusterization/component_connection.cu

```diff
@@ -0,0 +1,86 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s).
+#include "component_connection.hpp"
+#include "traccc/cuda/utils/definitions.hpp"
+
+namespace traccc::cuda {
+
+namespace kernel {
+__global__ void component_connection(
+    cell_container_types::const_view cells_view,
+    vecmem::data::vector_view<std::size_t> cluster_prefix_sum_view,
+    cluster_container_types::view clusters_view,
+    vecmem::data::jagged_vector_view<unsigned int> sparse_ccl_indices_view,
+    vecmem::data::vector_view<const device::prefix_sum_element_t>
+        cells_prefix_sum_view) {
+    auto idx = threadIdx.x + blockDim.x * blockIdx.x;
+
+    // Get device vector of the cells prefix sum
+    vecmem::device_vector<const device::prefix_sum_element_t> cells_prefix_sum(
+        cells_prefix_sum_view);
+    // Ignore if idx is out of range
+    if (idx >= cells_prefix_sum.size())
+        return;
+
+    // Get the indices for the module idx and the cell idx
+    auto module_idx = cells_prefix_sum[idx].first;
+    auto cell_idx = cells_prefix_sum[idx].second;
```

Comment:
The `auto` keyword can be pretty harmful to the readability of code, and I would argue that it's really not clear what the types of certain variables are. I'd prefer it if you would make some of these types concrete, as well as throughout the rest of the PR.